sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -272,6 +272,8 @@ def delete_table(
         try:
             self._client.table.delete(table=self.table_name, **kwargs)
         except HttpResponseError as error:
+            if error.status_code == 404:
```

Comment:
Could we add comments to all the docstrings that no error will be raised in the case of the target table/entity not being found?